lib/containers/urls.pm

```diff
@@ -51,64 +51,220 @@ sub get_opensuse_registry_prefix {
+            totest   => sub {
+                'registry.opensuse.org/' . get_opensuse_registry_prefix . 'opensuse/tumbleweed';
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x', 'arm']
+        },
+        '15.1' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.1' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.1/images/totest/containers/opensuse/leap:15.1';
+                } elsif (grep { $_ eq $arch } qw/aarch64 arm/) {
+                    'registry.opensuse.org/opensuse/leap/15.1/arm/images/totest/containers/opensuse/leap:15.1';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.2' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.2' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.2/images/totest/containers/opensuse/leap:15.2';
+                } elsif (grep { $_ eq $arch } qw/aarch64 arm/) {
+                    'registry.opensuse.org/opensuse/leap/15.2/arm/images/totest/containers/opensuse/leap:15.2';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.3' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.3' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3';
+                } elsif (grep { $_ eq $arch } qw/aarch64 arm/) {
+                    'registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        }
+    }
+);
+
+sub check_image_arch_in_registry {
+    my ($distri, $version, $arch) = @_;
+    (grep { $_ eq $arch } @{$registries{$distri}{$version}{available_arch}}) ? 1 : 0;
+}
+
 # Returns a tuple of image urls and their matching released "stable" counterpart.
 # If empty, no images available.
 sub get_suse_container_urls {
-    my $version    = shift // get_required_var('VERSION');
-    my $dotversion = $version =~ s/-SP/./r;                    # 15 -> 15, 15-SP1 -> 15.1
-    $dotversion = "${dotversion}.0" if $dotversion !~ /\./;    # 15 -> 15.0
-
+    my %args = (
+        version => get_required_var('VERSION'),
+        arch    => get_required_var('ARCH'),
+        distri  => get_required_var('DISTRI'),
+        @_
+    );
```

Comment:
My previous intention was to make this able to take specific parameters instead of variables as well.

WDYT @jlausuch?

Reply:
thats what it is, It is just a little bit obfuscated written. The code is more or less similar to:
```perl
    my (%args) = @_;
    $args{version} //= get_required_var('VERSION');
    $args{arch}    //= get_required_var('ARCH');
    $args{distri}  //= get_required_var('DISTRI');
```
with the exception, that the openqa variables are required, even if we call the function with parameters like `version => 'foo'`.

Reply:
OK! That is super cool!